lib/text/ssa_text_parser.js

```diff
@@ -0,0 +1,300 @@
+        const name = lineParts[1].trim();
+        const value = lineParts[2].trim();
+        if (name == 'Format') {
+          styleColumns = value.split(SsaTextParser.valuesFormat_);
+          continue;
+        }
+        if (name == 'Style') {
+          const values = value.split(SsaTextParser.valuesFormat_);
+          const style = {};
+          for (let c = 0; c < styleColumns.length && c < values.length; c++) {
+            style[styleColumns[c]] = values[c];
+          }
+          styles.push(style);
+          continue;
+        }
+      }
+    }
+
+    // Process cues
+    /** @type {!Array.<!shaka.extern.Cue>} */
+    const cues = [];
+
+    // Used to be able to iterate over the event parameters.
+    let eventColumns = null;
+
+    const eventLines = section.events.split(/\r?\n/);
+    for (const line of eventLines) {
+      if (/^\s*;/.test(line)) {
+        // Skip comment
+        continue;
+      }
+      const lineParts = SsaTextParser.lineParts_.exec(line);
+      if (lineParts) {
+        const name = lineParts[1].trim();
+        const value = lineParts[2].trim();
+        if (name == 'Format') {
+          eventColumns = value.split(SsaTextParser.valuesFormat_);
+          continue;
+        }
+        if (name == 'Dialogue') {
+          const values = value.split(SsaTextParser.valuesFormat_);
+          const data = {};
+          for (let c = 0; c < eventColumns.length && c < values.length; c++) {
+            data[eventColumns[c]] = values[c];
+          }
+
+          const startTime = SsaTextParser.parseTime_(data['Start']);
+          const endTime = SsaTextParser.parseTime_(data['End']);
+
+          // Note: Normally, you should take the "Text" field, but if it
+          // has a comma, it fails.
+          const indexOfPayload =
+            value.split(',', eventColumns.length - 1).join(',').length + 1;
+          const payload = value.substr(indexOfPayload)
+              .replace(/\\N/g, '\n') // '\n' for new line
+              .replace(/\{[^}]+\}/g, ''); // {\pos(400,570)}
+
+          const cue = new shaka.text.Cue(startTime, endTime, payload);
+
+          const styleName = data['Style'];
+          const styleData = styles.find((s) => s['Name'] == styleName);
+          if (styleData) {
+            SsaTextParser.addStyle_(cue, styleData);
+          }
+          cues.push(cue);
+          continue;
+        }
+      }
+    }
+
+    return cues;
+  }
+
+  /**
+   * Adds applicable style properties to a cue.
+   *
+   * @param {shaka.extern.Cue} cue
+   * @param {Object} style
+   * @private
+   */
+  static addStyle_(cue, style) {
+    const Cue = shaka.text.Cue;
+    const fontFamily = style['Fontname'];
+    if (fontFamily) {
+      cue.fontFamily = fontFamily;
+    }
+    const fontSize = style['Fontsize'];
+    if (fontSize) {
+      cue.fontSize = fontSize + 'px';
+    }
+    // TODO: convert a long integer BGR (blue-green-red) into RGB
+    // const color = style['PrimaryColour'];
```

Comment:
@TheModMaker I am not an expert in format conversion. Could you tell me how to do this (a snipet maybe)?

The specification says this: "A long integer BGR (blue-green-red) value. Ie. The byte order in the hexadecimal equivelent of this number is BBGGRR"

Value example: "&H00B4FCFC"

Reply:
The example you gave has a header "&H", followed by four one-byte hex values (00, B4, FC, FC).  I assume the last three are B, G, R.  Is the first value alpha?  Or ignored?

If it's ignored, this could be matched with a Regexp like:
```js
const match = /&H(..)(..)(..)(..)/.exec(input);
if (match) {
  cue.color = '#' + match[3] + match[2] + match[1];
}
```

This just flips the order of BGR to RGB and formats it as it would be in CSS (the format we expect in Cue).

You could also make the match more strict with something like:
```js
const match = /&H([0-9A-Fa-f])([0-9A-Fa-f])([0-9A-Fa-f])([0-9A-Fa-f])/.exec(input);
```

Reply:
Thank you very much for the help! I have already adapted your suggestion to work as expected.